### PR TITLE
Add midi upload to server

### PR DESCRIPTION
**Pull request by shilangyu**

### CODE REVIEW

**Review by piotrowskv — APPROVED** [review]

File: backend/app/db/init.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import sqlite3
+from pathlib import Path
+
+from app.db.repositories.training_sessions import TrainingSessionsRepository
+
+
+def create_database(directory: Path) -> tuple[TrainingSessionsRepository]:
+    assert not directory.exists() or directory.is_dir(
+    ), "Database path has to be a folder or not exist"
```
Comment (szgorski) [inline backend/app/db/init.py:9]: I'm not familiar with this annotation so leaving a comment to double-check if left on purpose

File: backend/app/app.py [hunk 2]
```diff
@@ -1,44 +1,39 @@
...
-        name='LSTM',
-        description='Sequential model generating each timestep one by one.')
-    MARKOV = m.ModelVariant(
-        id='e5893ba5-1cd1-4153-ac56-6b5898897503',
-        name='Markov Chain',
-        description='Statistical model generating most probable notes.')
-    GAN = m.ModelVariant(
-        id='eafdabd3-fe56-474d-91be-7a9eeeed2124',
-        name='GAN',
-        description='Generative model generating the whole song at once.')
```
Comment (szgorski) [inline backend/app/app.py:30]: nice to have it out of app.py :>

File: backend/app/db/repositories/training_sessions.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                   REFERENCES {self.SESSIONS_TABLE_NAME} (session_id)
+            )""")
+
+    async def register_session(self, model_id: str, files: list[UploadFile]) -> str:
+        """
+        Registers a new training session given a model id and files to train on.
+
+        Returns
+        ---
+        Returns a token identifying this session.
```
Comment (szgorski) [inline backend/app/db/repositories/training_sessions.py:36]: we may try to unify type/return annotations later if time will be on our side

File: models/setup.py [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 from sys import platform
 
 from setuptools import setup
 
-extra_deps = ['tensorflow-macos==2.10.0', 'tensorflow-metal==0.6.0'] if platform == "darwin" else ['tensorflow==2.10.0']
+extra_deps = ['tensorflow-macos==2.10.0',
+              'tensorflow-metal==0.6.0'] if platform == "darwin" else ['tensorflow==2.10.0']
 
-midi_dir = Path(__file__).resolve().parent.parent.joinpath('midi')
+# ../midi is not listed as requirements! It is the responsibility of `models` dependents to also install `../midi`
```
Comment (szgorski) [inline models/setup.py]: **dependents** or **dependencies**?
tbh didn't know _dependents_ is a valid word

**Review by szgorski — APPROVED** [review]

File: models/setup.py (hunk 4 above)
Comment (shilangyu) [reply models/setup.py]: I did mean _dependents_ (those that depend on). I will change the wording here.

**Pull request merged**